components/molecules/PhoneNumberInput.vue

```diff
@@ -0,0 +1,298 @@
+import CountrySelectorIndex from '../atoms/PhoneNumberInput/CountrySelector/CountrySelectorIndex.vue'
+import PhoneNumberInput from '~/components/molecules/PhoneNumberInput.vue'
+
+const phoneNumberInputEl = ref<typeof PhoneNumberInput | undefined>(undefined)
+
+const browserLocale = () => {
+    if (!window) return null
+    const browserLocale = window.navigator.language || window.navigator.language
+    let locale = browserLocale ? browserLocale.substring(3, 4).toUpperCase() : null
+    if (locale === '') locale = browserLocale.substring(0, 2).toUpperCase()
+    return locale
+}
+
+const isCountryAvailable = (locale) => {
+    return countriesIso.includes(locale)
+}
+
+const props = withDefaults(
+    defineProps<{
+        value: string
+        id: string
+        color: string
+        errorColor: string
+        defaultCountryCode: string | null
+        preferredCountries: string[]
+        onlyCountries: string[]
+        ignoredCountries: string[]
+        translations: {} | null
+        noValidatorState: boolean
+        error: boolean
+        noExample: boolean
+        countriesHeight: number
+        noUseBrowserLocale: boolean
+        fetchCountry: boolean
+        borderRadius: number
+    }>(),
+    {
+        value: '',
+        id: 'MazPhoneNumberInput',
+        color: 'dodgerblue',
+        errorColor: 'orangered',
+        defaultCountryCode: null,
+        preferredCountries: null,
+        onlyCountries: () => [''],
+        ignoredCountries: () => [''],
+        translations: null,
+        noValidatorState: false,
+        error: false,
+        noExample: false,
+        countriesHeight: 30,
+        noUseBrowserLocale: false,
+        fetchCountry: false,
+        borderRadius: 4,
+    },
+)
+
+const emit = defineEmits(['update', 'input', 'phone-number-focused', 'phone-number-blur'])
+const results = ref({ countryCode: undefined, formatInternational: undefined, isValid: undefined })
+const userLocale = ref(props.defaultCountryCode)
+const lastKeyPressed = ref(0)
+const phoneNumber = ref<string[] | string | number[] | null | undefined>('')
+
+const uniqueId = computed(() => `${props.id}-${Math.random().toString(36).substring(2, 9)}`)
+
+const translatedCountryName = computed(() => ({
+    ...locales,
+    ...props.translations,
+}))
+
+const codesCountries = computed(() => countries)
+
+const countryCode = computed({
+    get() {
+        return userLocale.value || results.value.countryCode
+    },
+    set(newCountry) {
+        setLocale(newCountry)
+        phoneNumberInputEl.value?.$el.querySelector('input').focus()
+    },
+})
+
+const shouldChooseCountry = computed(() => {
+    return !countryCode.value && !!phoneNumber.value
+})
+
+const isValid = computed(() => results.value.isValid)
+
+const phoneNumberExample = computed(() => {
+    const exampleNumber = countryCode.value ? getExampleNumber(countryCode.value, examples) : null
+    return exampleNumber ? exampleNumber.formatNational() : null
```

Comment:
can just be optional `return exampleNumber?.formatNational() ?? null` if you must return null